src/io/syncsocket.c

```diff
@@ -228,27 +230,63 @@ static void gc_free(MVMThreadContext *tc, MVMObject *h, void *d) {
     MVM_free(data);
 }
 
+static size_t get_struct_size_for_family(sa_family_t family) {
+    switch (family) {
+        case AF_INET6:
+            return sizeof(struct sockaddr_in6);
+        case AF_INET:
+            return sizeof(struct sockaddr_in);
+#ifndef _WIN32
+        case AF_UNIX:
+            return sizeof(struct sockaddr_un);
+#endif
+        default:
+            return sizeof(struct sockaddr);
+    }
+}
+
 /* Actually, it may return sockaddr_in6 as well; it's not a problem for us, because we just
  * pass is straight to uv, and the first thing it does is it looks at the address family,
  * but it's a thing to remember if someone feels like peeking inside the returned struct. */
 struct sockaddr * MVM_io_resolve_host_name(MVMThreadContext *tc, MVMString *host, MVMint64 port) {
     char *host_cstr = MVM_string_utf8_encode_C_string(tc, host);
     struct sockaddr *dest;
-    struct addrinfo *result;
     int error;
+    int flags = 0;
+    struct addrinfo *result;
+#ifndef _WIN32
+    struct sockaddr_un *result_un = MVM_malloc(sizeof(struct sockaddr_un));
```

Comment:
`result_un` is malloced and never freed.
Data is copied into it, then copied into a (newly malloc-ed) `dest`, and `dest` is returned, and `result_un` is thrown away.
This seems silly.
It seems to me that `result_un` should be declared (and malloced) _inside_ of the `if (flags & AF_UNIX) {` block, and returned _instead_ of `dest`.